From 9873830ef33b03ed835324564c89d6b288e072a0 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@foss.st.com>
Date: Mon, 17 Jun 2024 16:59:05 +0200
Subject: [PATCH 621/841] feat(stm32mp1-fdts): add DDR power supplies
 description

Describe DDR power supplies for stm32mp1 boards.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git f16830aadae1e1ae54830e4c46ad1da510a4b298 feat(stm32mp1-fdts): add DDR power supplies description]
Change-Id: I56f31502b830c082ec660815dfc0dc1893559359
Signed-off-by: Pascal Paillet <p.paillet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/388116
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
Tested-by: Pascal PAILLET <p.paillet@st.com>
---
 fdts/stm32mp135f-dk.dts           | 5 +++++
 fdts/stm32mp157a-avenger96.dts    | 5 +++++
 fdts/stm32mp157a-ed1.dts          | 6 ++++++
 fdts/stm32mp157c-ed1.dts          | 6 ++++++
 fdts/stm32mp157c-odyssey-som.dtsi | 6 ++++++
 fdts/stm32mp157d-ed1.dts          | 6 ++++++
 fdts/stm32mp157f-ed1.dts          | 6 ++++++
 fdts/stm32mp15xx-dhcom-som.dtsi   | 6 ++++++
 fdts/stm32mp15xx-dhcor-som.dtsi   | 6 ++++++
 fdts/stm32mp15xx-dkx.dtsi         | 6 ++++++
 fdts/stm32mp15xx-osd32.dtsi       | 6 ++++++
 11 files changed, 64 insertions(+)

diff --git a/fdts/stm32mp135f-dk.dts b/fdts/stm32mp135f-dk.dts
index fb5802ab9..22f7bb2d5 100644
--- a/fdts/stm32mp135f-dk.dts
+++ b/fdts/stm32mp135f-dk.dts
@@ -60,6 +60,11 @@
 	cpu-supply = <&vddcpu>;
 };
 
+&ddr {
+	vdd-supply = <&vdd_ddr>;
+	vref-supply = <&vref_ddr>;
+};
+
 &hash {
 	status = "okay";
 };
diff --git a/fdts/stm32mp157a-avenger96.dts b/fdts/stm32mp157a-avenger96.dts
index 73abe5607..f62387e67 100644
--- a/fdts/stm32mp157a-avenger96.dts
+++ b/fdts/stm32mp157a-avenger96.dts
@@ -35,6 +35,11 @@
 	};
 };
 
+&ddr {
+	vdd-supply = <&vdd_ddr>;
+	vref-supply = <&vref_ddr>;
+};
+
 &i2c4 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c4_pins_a>;
diff --git a/fdts/stm32mp157a-ed1.dts b/fdts/stm32mp157a-ed1.dts
index 18334983b..ba3392d67 100644
--- a/fdts/stm32mp157a-ed1.dts
+++ b/fdts/stm32mp157a-ed1.dts
@@ -49,6 +49,12 @@
 	cpu-supply = <&vddcore>;
 };
 
+&ddr {
+	vdd-supply = <&vdd_ddr>;
+	vtt-supply = <&vtt_ddr>;
+	vref-supply = <&vref_ddr>;
+};
+
 &hash1 {
 	status = "okay";
 };
diff --git a/fdts/stm32mp157c-ed1.dts b/fdts/stm32mp157c-ed1.dts
index ba2fd1f14..dc4939398 100644
--- a/fdts/stm32mp157c-ed1.dts
+++ b/fdts/stm32mp157c-ed1.dts
@@ -53,6 +53,12 @@
 	status = "okay";
 };
 
+&ddr {
+	vdd-supply = <&vdd_ddr>;
+	vtt-supply = <&vtt_ddr>;
+	vref-supply = <&vref_ddr>;
+};
+
 &hash1 {
 	status = "okay";
 };
diff --git a/fdts/stm32mp157c-odyssey-som.dtsi b/fdts/stm32mp157c-odyssey-som.dtsi
index 785a73ff0..9866cb499 100644
--- a/fdts/stm32mp157c-odyssey-som.dtsi
+++ b/fdts/stm32mp157c-odyssey-som.dtsi
@@ -50,6 +50,12 @@
 	status = "okay";
 };
 
+&ddr {
+	vdd-supply = <&vdd_ddr>;
+	vtt-supply = <&vtt_ddr>;
+	vref-supply = <&vref_ddr>;
+};
+
 &hash1 {
 	status = "okay";
 };
diff --git a/fdts/stm32mp157d-ed1.dts b/fdts/stm32mp157d-ed1.dts
index 0693d72ab..32214d640 100644
--- a/fdts/stm32mp157d-ed1.dts
+++ b/fdts/stm32mp157d-ed1.dts
@@ -49,6 +49,12 @@
 	cpu-supply = <&vddcore>;
 };
 
+&ddr {
+	vdd-supply = <&vdd_ddr>;
+	vtt-supply = <&vtt_ddr>;
+	vref-supply = <&vref_ddr>;
+};
+
 &hash1 {
 	status = "okay";
 };
diff --git a/fdts/stm32mp157f-ed1.dts b/fdts/stm32mp157f-ed1.dts
index d097e0568..88fdf9fb2 100644
--- a/fdts/stm32mp157f-ed1.dts
+++ b/fdts/stm32mp157f-ed1.dts
@@ -53,6 +53,12 @@
 	status = "okay";
 };
 
+&ddr {
+	vdd-supply = <&vdd_ddr>;
+	vtt-supply = <&vtt_ddr>;
+	vref-supply = <&vref_ddr>;
+};
+
 &hash1 {
 	status = "okay";
 };
diff --git a/fdts/stm32mp15xx-dhcom-som.dtsi b/fdts/stm32mp15xx-dhcom-som.dtsi
index 19c4fcd0e..2c10a14e1 100644
--- a/fdts/stm32mp15xx-dhcom-som.dtsi
+++ b/fdts/stm32mp15xx-dhcom-som.dtsi
@@ -32,6 +32,12 @@
 	cpu-supply = <&vddcore>;
 };
 
+&ddr {
+	vdd-supply = <&vdd_ddr>;
+	vtt-supply = <&vtt_ddr>;
+	vref-supply = <&vref_ddr>;
+};
+
 &hash1 {
 	status = "okay";
 };
diff --git a/fdts/stm32mp15xx-dhcor-som.dtsi b/fdts/stm32mp15xx-dhcor-som.dtsi
index f17c3ce39..877b032a4 100644
--- a/fdts/stm32mp15xx-dhcor-som.dtsi
+++ b/fdts/stm32mp15xx-dhcor-som.dtsi
@@ -27,6 +27,12 @@
 	cpu-supply = <&vddcore>;
 };
 
+&ddr {
+	vdd-supply = <&vdd_ddr>;
+	vtt-supply = <&vtt_ddr>;
+	vref-supply = <&vref_ddr>;
+};
+
 &hash1 {
 	status = "okay";
 };
diff --git a/fdts/stm32mp15xx-dkx.dtsi b/fdts/stm32mp15xx-dkx.dtsi
index e75bfaee9..ea346ace6 100644
--- a/fdts/stm32mp15xx-dkx.dtsi
+++ b/fdts/stm32mp15xx-dkx.dtsi
@@ -47,6 +47,12 @@
 	cpu-supply = <&vddcore>;
 };
 
+&ddr {
+	vdd-supply = <&vdd_ddr>;
+	vtt-supply = <&vtt_ddr>;
+	vref-supply = <&vref_ddr>;
+};
+
 &hash1 {
 	status = "okay";
 };
diff --git a/fdts/stm32mp15xx-osd32.dtsi b/fdts/stm32mp15xx-osd32.dtsi
index c5c04f3da..859da3e8f 100644
--- a/fdts/stm32mp15xx-osd32.dtsi
+++ b/fdts/stm32mp15xx-osd32.dtsi
@@ -6,6 +6,12 @@
 
 #include "stm32mp15-pinctrl.dtsi"
 
+&ddr {
+	vdd-supply = <&vdd_ddr>;
+	vtt-supply = <&vtt_ddr>;
+	vref-supply = <&vref_ddr>;
+};
+
 &i2c4 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c4_pins_a>;
-- 
2.47.2

